### ISSUE TITLE
Changing accounts on MetaMask wallet doesn't prompt you to revalidate session

### ISSUE THREAD

**Issue by tylerpashigian** [spruceid/ssx#79]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When connecting to a MetaMask wallet via rainbowkit/siwe, I am prompted to sign a message to validate my session. When I switch accounts from within the same MetaMask wallet, the `address` from wagmi's `useAccount()` hook is updated, but the jwt associated with the session from the `useSession()` next-auth/react hook is tied to my previous address.

### Expected Behavior

I would like the session to be associated with the new account, even if it means signing a new signature. Right now I am manually flushing the session and programmatically logging in in a React `useEffect()`.

### Steps To Reproduce

1. Connect MetaMask wallet
2. Sign in with wallet
3. Change accounts in wallet
4. Observe jwt in session is previous wallet address

**Comment by obstropolos** [comment]
Noting - working through cases of this in #88

**Comment by franklovefrank** [comment]
@tylerpashigian hi tyler, do you mind sharing more details about the use case that prompted this issue? particularly curious how you're using and checking the jwt

**Comment by franklovefrank** [comment]
@tylerpashigian Hi Yyler, do you mind sharing more details about the use case that prompted this issue? Particularly curious how you're using and checking the JWT

**Comment by franklovefrank** [comment]
@tylerpashigian Hi Tyler, do you mind sharing more details about the use case that prompted this issue? Particularly curious how you're using and checking the JWT

**Comment by tylerpashigian** [comment]
> @tylerpashigian Hi Tyler, do you mind sharing more details about the use case that prompted this issue? Particularly curious how you're using and checking the JWT

Hey @franklovefrank, sure. So we have a Hasura db connection and are passing the `session.token.accessToken` that is generated in the `[...nextAuth.ts]` file (we have jwt as our session strategy). When signing in initially, this accessToken is generated with an associated wallet address, `session.token.sub`. The issue is when changing wallets within metamask, the address from the WAGMI `useAccount()` hook changes, but the `session.token.sub`/`session.token.accessToken` are still associated with the existing address, so the auth status isnt valid.

I created a workaround for this for now, which is just a simple force `signOut` when the address changes, which forces users to sign a new message with the newly connected wallet. We wanted to maintain the auth state when refreshing the page (as long as the accessToken or refreshToken were still valid). To prevent sign outs when the address changes from null to the connected wallet (while the hook initializes during a refresh or initial page load), I am storing a `loaded` var and a `prevAddress` var in a redux store synced with our localstorage and checking to see if the last signed in wallet address is different than the previously connected wallet, only after the `prevAddress` has been loaded correctly from our localstorage to our redux store.

I was hoping that this would be managed automatically and changing wallets would prompt users to sign a new message with the newly connected wallet.

```
import { useSession, signOut } from 'next-auth/react';
import { useAccount, useDisconnect, useSignMessage } from 'wagmi';

const { address, isConnected } = useAccount();

const prevAddress: string = useSelector((state: any) => state.account.address);
const loaded: boolean = useSelector((state: any) => state.account.loaded);

useEffect(() => {
  const connected = address && isConnected;
  if (
    (connected && prevAddress !== address && loaded) ||
    !!session?.token.error
  ) {
    signOut({ redirect: false });
  }

  if (connected) {
    dispatch(updateAddress(address));
  }
}, [address, dispatch, isConnected, loaded, prevAddress, session]);
  ```

**Comment by obstropolos** [comment]
Updates to this were included in [ssx-react/v1.3.0](https://github.com/spruceid/ssx/releases/tag/ssx-react-v1.3.0)
